Yosemite/YosemiteTests/Mocks/MockStorageManager.swift

```diff
@@ -87,16 +87,36 @@ public class MockStorageManager: StorageManagerType {
     /// Handles a write operation using the background context and saves changes when done.
     /// Using view storage to write for simplicity in tests
     ///
-    public func performAndSave(_ closure: @escaping (StorageType) -> Void, completion: (() -> Void)?, on queue: DispatchQueue) {
+    public func performAndSave(_ operation: @escaping (StorageType) -> Void,
+                               completion: (() -> Void)?,
+                               on queue: DispatchQueue) {
         let context = persistentContainer.viewContext
         context.performAndWait {
-            closure(context)
+            operation(context)
             context.saveIfNeeded()
             queue.async {
                 completion?()
             }
         }
     }
+
+    /// Handles a write operation using the background context and saves changes and returns result when done.
+    /// Using view storage to write for simplicity in tests.
+    ///
+    public func performAndSave<T>(_ operation: @escaping (StorageType) throws -> T,
+                                  completion: @escaping (Result<T, Error>) -> Void,
+                                  on queue: DispatchQueue) {
+        let context = persistentContainer.viewContext
+        context.performAndWait {
+            let result = Result(catching: { try operation(context) })
+            if case .success = result {
+                context.saveIfNeeded()
+            }
```

Comment:
Same [as above](https://github.com/woocommerce/woocommerce-ios/pull/14099/files#r1797912567), I let you decide how much we want to log failures if these happen, or just let them propagate. Could be entirely pointless to handle the error at this point 👍

Reply:
I don't think logging failures is necessary for this mock class. I hope it makes sense!